tianshou/policy/multiagent/mapolicy.py

```diff
@@ -36,7 +36,8 @@ def process_fn(self, batch: Batch, buffer: ReplayBuffer,
         # reward can be empty Batch (after initial reset) or nparray.
         has_rew = isinstance(buffer.rew, np.ndarray)
         if has_rew:  # save the original reward in save_rew
-            save_rew, buffer.rew = buffer.rew, Batch()
+            # since buffer.__setattr__ is not override, here we use _meta
+            save_rew, buffer._meta.rew = buffer.rew, Batch()
```

Comment:
Here is a bug fix. `buffer.rew = Batch` will not change the reward saved previously.

Reply:
good point.